addons/binding/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/handler/HeosBridgeHandler.java

```diff
@@ -0,0 +1,554 @@
+import org.openhab.binding.heos.internal.discovery.HeosPlayerDiscovery;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandler;
+import org.openhab.binding.heos.internal.resources.HeosEventListener;
+import org.openhab.binding.heos.internal.resources.HeosGroup;
+import org.openhab.binding.heos.internal.resources.HeosPlayer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeosSystemHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Johannes Einig - Initial contribution
+ */
+public class HeosBridgeHandler extends BaseBridgeHandler implements HeosEventListener, DiscoveryListener {
+
+    private List<String> heosPlaylists = new ArrayList<>();
+
+    private Map<ThingUID, ThingHandler> handlerList = new HashMap<>();
+    private Map<String, String> selectedPlayer = new HashMap<>();
+    private List<String[]> selectedPlayerList = new ArrayList<>();
+    private Map<ThingUID, ThingStatus> thingOnlineState = new HashMap<>();
+    private HeosChannelHandlerFactory channelHandlerFactory;
+
+    private ScheduledExecutorService initPhaseExecutor;
+    private InitProcedure initPhaseRunnable = new InitProcedure();
+
+    private HeosPlayerDiscovery playerDiscovery;
+    private HeosSystem heos;
+    private HeosFacade api;
+
+    private int heartbeatPulse = 0;
+
+    private boolean isRegisteredForChangeEvents = false;
+    private boolean bridgeIsConnected = false;
+    private boolean handleGroups = true;
+    private boolean loggedIn = false;
+    private boolean connectionDelay = false;
+    private boolean bridgeHandlerdisposalOngoing = false;
+
+    private Logger logger = LoggerFactory.getLogger(HeosBridgeHandler.class);
+
+    public HeosBridgeHandler(Bridge thing, HeosSystem heos, HeosFacade api) {
+        super(thing);
+        this.heos = heos;
+        this.api = api;
+        channelHandlerFactory = new HeosChannelHandlerFactory(this, api);
+    }
+
+    @Override
+    public void handleCommand(@NonNull ChannelUID channelUID, @NonNull Command command) {
+        if (command instanceof RefreshType) {
+            return;
+        }
+        HeosChannelHandler channelHandler = channelHandlerFactory.getChannelHandler(channelUID);
+        if (channelHandler != null) {
+            channelHandler.handleCommand(command, this, channelUID);
+            return;
+        }
+    }
+
+    public void resetPlayerList(@NonNull ChannelUID channelUID) {
+        for (int i = 0; i < selectedPlayerList.size(); i++) {
+            updateState(selectedPlayerList.get(i)[1], OnOffType.OFF);
+        }
+        selectedPlayerList.clear();
+        updateState(channelUID, OnOffType.OFF);
+    }
+
+    @Override
+    public synchronized void initialize() {
+        if (bridgeIsConnected) {
+            return;
+        }
+        loggedIn = false;
+
+        logger.info("Initialize Bridge '{}' with IP '{}'", thing.getConfiguration().get(NAME),
```

Comment:
Please only use DEBUG level for such logs (same for other places throughout this binding), see https://www.openhab.org/docs/developer/development/guidelines.html#e-logging.